src/data/badge.ts

```diff
@@ -0,0 +1,28 @@
+import type { Localizable } from "#app/interfaces/locales";
+import type { BadgeType } from "#enums/badge-type";
+import type { Type } from "#enums/type";
+
+/**
+ * Badges are key items that are given to the player and affect the run
+ *
+ * @param id the corresponding BadgeType enum
+ * @param name the name of the badge
+ * @param effect the description of what the badge does
+ * @param generation which generation the badge originated
+ * @param type the pokemon type associated with the badge
+ * @param powerLevel a way to tweak how effective a badge is
+ */
+export abstract class Badge implements Localizable {
```

Comment:
Please don't implement `Localizable` anymore. It's an anti pattern!

```suggestion
export abstract class Badge {
```

Reply:
I'm not entirely sure how this would be replaced for the existing ones though.